mifospay/src/main/java/org/mifos/mobilewallet/mifospay/savedcards/ui/AddCardDialog.java

```diff
@@ -51,6 +56,23 @@
     Button btnCancel;
     CardsContract.CardsPresenter mCardsPresenter;
     private BottomSheetBehavior mBottomSheetBehavior;
+    @BindViews({R.id.til_fName, R.id.til_lName, R.id.til_card_number, R.id.til_cvv})
+    List<TextInputLayout> mTextInputLayouts;
+    private boolean fieldsValid;
+    private final ButterKnife.Action<TextInputLayout> CHECK_ERROR =
+            new ButterKnife.Action<TextInputLayout>() {
+                @Override
+                public void apply(@NonNull TextInputLayout view, int index) {
+                    EditText editText = view.getEditText();
+                    if (editText == null || editText.getEditableText() == null
+                            || editText.getEditableText().toString().isEmpty()) {
```

Comment:
Add .trim() method to prevent names containing only whitespaces.

Reply:
Done.